include/enum/boundary_type.hpp

```
@@ -0,0 +1,63 @@
+/**
+ * @file boundary_type.hpp
+ * @brief Definition and utilities for BoundaryType enumeration.
+ * @details
+ * This file contains the definition of the BoundaryType enumeration,
+ * along with utility functions for converting between the enum and its
+ * string representation, as well as an overloaded output stream operator.
+ */
+#ifndef NM4PDE_BOUNDARY_TYPE_HPP
+#define NM4PDE_BOUNDARY_TYPE_HPP
+#include <stdexcept>
+
+/**
+ * @brief Enumeration for different types of problems.
+ */
+enum class BoundaryType : char { Zero, MMS, Expr };
+
+/**
+ * @brief Convert BoundaryType enum to string for output.
+ * @param boundary_type BoundaryType enum value.
+ * @return Corresponding string representation.
+ */
+inline std::string to_string(const BoundaryType boundary_type) {
+    switch (boundary_type) {
+        case BoundaryType::Zero:
+            return "zero";
+        case BoundaryType::MMS:
+            return "mms";
+        case BoundaryType::Expr:
+            return "expr";
+        default:
+            return "unknown";
+    }
+}
```

You might want to check out [magic_enum](https://github.com/Neargye/magic_enum), I've never used it personally but it's supposed to automate this kind of thing. Also, it is bundled with `deal.II` (although I'm not sure in which version this started - I think 9.6 or 9.7).

Also, it might be a better idea to raise an exception upon encountering an unknown value, rather than returning a placeholder string.